New feature/context page (#7)
RIT-EVT/telemetry-database: README.md

Currently to use the script run main and when prompted input the path to the .mf4 file you want to have uploaded.<br>

As we do not have means to collect context data yet, we use a dummy context with an id of 1. This is gross and bad. We will be removing that as soon as we have a way to query the user for context data.<br>

## Web UI 

> This react-strap based website is responsible for gathering all the context relating to the data we gather.<br>

Reviewer: This is going to be a single page of many in this website which collects more than just the context.

> This react-strap based website is responsible for gathering all context and data relating to the bikes and all events<br>

Reviewer 2: ```suggestion
This react-strap-based website allows users to input context and configuration information and to upload CAN data files.<br>
```

> Ideally, mechies will just download the files and run it on their computer. This includes running both the website and backend api that goes along with it.<br>

Reviewer 3: I dont like how mece is spelled here. MECE's, or mech-e's, or something. idk. rn its off putting. Im grammar policing you frfr

In the future, I hope to host at least the backend restful api, and eventually the website itself, on a web server, but that is not going to happen for a long time. To anyone that works on this in the future, if this message is still here, that means I never went back and fixed the mess I made, and for that I sincerely apologize for. This was meant to be quick and dirty, but I attempted to follow best practices and self document the code where I could. <br>

## RESTful API

> The RESTful API is built off python using Flask. <br>

Reviewer 2: ```suggestion
The RESTful API is built with Python using Flask. <br>
```

Someday, this will be hosted on a central server, but for now it runs locally. To run this application, see directions below under **Running the Server**. <br>

For any future developers, standard practices should be followed when building this API. This includes separating calls into different classes and using the `GET`, `PUT`, `POST`, and `DELETE` calls correctly. Additionally, any changes to the url path for a call should be updated in the ServerPath.json file.

### Download Instructions

#### Node JS

> The first step is to install Node JS if you do not already have it installed.<br>

Reviewer 2: If you can make it, it's really nice to have a list of commands that can be copied instead of going to specific websites to read about how to install things. If it doesn't work in this case, that's okay, but it would be nice.

Also, the explanations you have here can be helpful, but it's annoying to dig through a bunch of text to find the commands. Maybe group commands into sections and then explain a bunch at once instead of explaining each command individually

Author: Grouped commands for installing Node JS and other commands throughout the file

1. Follow the download instructions found here [Node JS](https://nodejs.org/en/download/package-manager)
2. Verify download by running `npm --version` or `node -v`<br>

> After Node JS has been installed, clone this repo onto your computer. If you have installed, open the project in VS Code. If you don't have in installed, just follow along in the command window. <br>

Reviewer 2: ```suggestion
After Node JS has been installed, clone this repo onto your computer. If you have it installed, you can use VS Code to open the project. If you don't have it installed, just follow along in the command window. <br>
```

Also run `npm install react` and `npm install reactstrap`

> 

Reviewer: Please add a note about getting/updating Python. Most people dont even know what a PATH variable is let alone how to set python in theirs.

#### **Running the Website**

In the terminal, `cd` into the `~...\Telemetry-Database\TelemetrySite\client` folder and run the command `npm install`. <br>

Once that has finished running, run the command `npm start`. <br>

>  If your website didn't start, check that you are in the correct folder and you installed Node JS. <br>

Reviewer 2: ```suggestion
If your website didn't start, check that you are in the correct folder and you installed Node JS. <br>
```

After your sever is online, make sure you keep that terminal open. **Closing that terminal for any reason will kill your website**.<br>

#### **Running the Server**

Open a new terminal and `cd` into the `~...\Telemetry-Database` folder. <br>

> Run the command `pip instal -r requirements.txt`. <br>

Reviewer 2: ```suggestion
Run the command `pip install -r requirements.txt`. <br>
```

If `pip` isn't being recognized, run `python -V`. If python isn't recognized, download the installer here [Python](https://www.python.org/downloads/). Python should add itself as environmental variable automatically, but you may need to click **Add As Environmental Variable** on the last page before you close the installer.<br>

In your terminal `cd` into the `~...\Telemetry-Database\TelemetrySite\server` folder and run `python .\server.py`

#### **.env File**

In order to connect to the database, you must contact one of the database leads for the .env file.<br> 

Once they have sent it to you, place it in the `~...\Telemetry-Database` folder in a file names **credentials.env**.

> **WARNING**

> >**.env files should never be shared or the contents sent anywhere without the Firmware Team's permission**

Reviewer 2: ```suggestion
>**.env files should never be shared or the contents sent anywhere without the Firmware Team's permission**
>**DO NOT ADD credentials.env TO GIT**
```

> 

Reviewer 2: ```suggestion
```
